server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/OdcDBStructureComparator.java

```diff
@@ -0,0 +1,146 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.structurecompare;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.compress.utils.Lists;
+
+import com.oceanbase.odc.core.shared.constant.DialectType;
+import com.oceanbase.odc.service.plugin.SchemaPluginUtil;
+import com.oceanbase.odc.service.structurecompare.comparedbobject.TableStructureComparator;
+import com.oceanbase.odc.service.structurecompare.model.ComparisonResult;
+import com.oceanbase.odc.service.structurecompare.model.DBObjectComparisonResult;
+import com.oceanbase.odc.service.structurecompare.model.DBStructureComparisonConfig;
+import com.oceanbase.tools.dbbrowser.editor.DBTableEditor;
+import com.oceanbase.tools.dbbrowser.model.DBObjectType;
+import com.oceanbase.tools.dbbrowser.model.DBTable;
+import com.oceanbase.tools.dbbrowser.model.DBTable.DBTableOptions;
+import com.oceanbase.tools.dbbrowser.model.DBTableColumn;
+import com.oceanbase.tools.dbbrowser.model.DBTableConstraint;
+import com.oceanbase.tools.dbbrowser.model.DBTableIndex;
+import com.oceanbase.tools.dbbrowser.schema.DBSchemaAccessor;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author jingtian
+ * @date 2024/1/4
+ * @since ODC_release_4.2.4
+ */
+@Slf4j
+public class OdcDBStructureComparator implements DBStructureComparator {
+    private final List<DialectType> supportedDialectTypes =
+            Arrays.asList(DialectType.MYSQL, DialectType.OB_MYSQL, DialectType.OB_ORACLE);
+    private final List<DBObjectType> supportedDBObjectTypes = Arrays.asList(DBObjectType.TABLE);
+
+    @Override
+    public List<DBObjectComparisonResult> compare(@NonNull DBStructureComparisonConfig srcConfig,
+            @NonNull DBStructureComparisonConfig tgtConfig)
+            throws SQLException {
+        List<DBObjectComparisonResult> returnVal = new ArrayList<>();
+        checkConfig(srcConfig, tgtConfig);
+
+        DBSchemaAccessor srcAccessor = SchemaPluginUtil.getSchemaBrowserExtension(srcConfig.getDialectType())
+                .getDBSchemaAccessor(srcConfig.getDataSource().getConnection());
+        DBSchemaAccessor tgtAccessor = SchemaPluginUtil.getSchemaBrowserExtension(tgtConfig.getDialectType())
+                .getDBSchemaAccessor(tgtConfig.getDataSource().getConnection());
+        DBTableEditor tgtTableEditor = SchemaPluginUtil.getSchemaBrowserExtension(tgtConfig.getDialectType())
+                .getDBTableEditor(tgtConfig.getDataSource().getConnection());
+
+        Map<String, DBTable> srcTables = buildSchemaTables(srcAccessor, srcConfig.getSchemaName());
```

Comment:
When naming a map, better follow the rule: key2Value~

Reply:
done